### ISSUE TITLE
[v2] 2.0.0rc2 refuses to manage ingress in namespaces with long names

### ISSUE THREAD

**Issue by postmaxin** [kubernetes-sigs/aws-alb-ingress-controller#1463]
Hi,
   I'm testing out 2.0.0rc2 as we really need the ALB aggregation functionality.

   It appears that the reconciler can not handle namespace names that are within Kubernetes' 63-character limit. This seems to be because they are trying to include the namespace name in a much longer name. For example, for the namespace "b-bugfix-jh-dt-3120-clean-jenkins-run" (which isn't that long of a name, to be honest), it is prepending "alb.ingress.k8s.aws/" and suffixing the ingress name, ".ingress-external-dex-auth" onto another k8s object with a 63-character name limit, causing a crash.

    There has to be some way to label these things that doesn't involve concatenation. We already truncate our namespace names at 63 characters when created (they are the names of branches in git in our CI/CD pipeline):

```
{"level":"error","ts":1601497876.3211784,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"ingress","name":"rex-ingress-external-dex-auth","namespace":"b-bugfix-jh-dt-3120-clean-jenkins-run","error":"Ingress.extensions \"ingress-external-dex-auth\" is invalid: metadata.finalizers: Invalid value: \"alb.ingress.k8s.aws/b-bugfix-jh-dt-3120-clean-jenkins-run.ingress-external-dex-auth\": name part must be no more than 63 characters","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90"}
```

**Comment by M00nF1sh** [comment]
/kind bug
thanks for reporting this. 
we used `namespace/IngressName` as the implicit groupID for Ingresses without `group.name` annotation. and it's been used a finalizer to allow us track the "groups" an Ingress belongs to.(and we use such information to decide whether there are left resources to cleanup).

need to find a way to make long names work.

**Comment by M00nF1sh** [comment]
I think we can use different finalizer format for explicitIngressGroup and implicitIngressGroup.  e.g.
explicit group: `group.ingress.k8s.aws/group-name`
implicit group: `ingress.k8s.aws`
For implicitGroup, it's groupID can be inferred from Ingress namespace and name.

**Comment by postmaxin** [comment]
If I understand correctly, that would mean that group names will be limited
to 42 characters, correct?
Not a problem for me, just want to call out the limitation.

**Comment by M00nF1sh** [comment]
@postmaxin the prefixes is not counted in 63 length limit, we can still have a groupName of 63 length limit not matter what prefix is used.

**Comment by M00nF1sh** [comment]
closing this since it's addressed by [v2.0.0-rc3 release](https://github.com/kubernetes-sigs/aws-alb-ingress-controller/releases/tag/v2.0.0-rc3) :D.
feel free to reopen if further issue exists.
